plugins/modules/cm_kerberos.py

```diff
@@ -0,0 +1,487 @@
+      - Changing this setting would clear up all existing credentials and keytabs from Cloudera Manager.
+    type: str
+    required: false
+  kdc_type:
+    description:
+      - Type of KDC Kerberos key distribution center (KDC) used for authentication.
+    type: str
+    required: false
+    default: present
+    choices:
+      - 'MIT KDC'
+      - 'Active Directory'
+      - 'Red Hat IPA'
+  kdc_admin_host:
+    description:
+      - KDC Admin Server Host
+      - Port number is optional and can be provided as V(hostname:port)
+    type: str
+    required: false
+  kdc_host:
+    description:
+      - KDC Server Host
+      - Port number is optional and can be provided as V(hostname:port)
+    type: str
+    required: false
+  krb_auth_enable:
+    description:
+      - Enable SPNEGO/Kerberos Authentication for the Admin Console and API
+    type: bool
+    required: false
+  ad_account_prefix:
+    description:
+      - Prefix used in names while creating accounts in Active Directory.
+      - The prefix can be up to 15 characters long and can be set to identify accounts used for authentication by CDH processes.
+      - Used only if O(kdc_type='Active Directory').
+    type: str
+    required: false
+  ad_kdc_domain:
+    description:
+      - Active Directory suffix where all the accounts used by CDH daemons will be created.
+      - Used only if O(kdc_type='Active Directory').
+    type: str
+    required: false
+  ad_delete_on_regenerate:
+    description:
+      - Active Directory Delete Accounts on Credential Regeneration.
+      - Set this option to V(true) if regeneration of credentials should automatically delete the associated Active Directory accounts.
+      - Used only if O(kdc_type='Active Directory').
+    type: bool
+    required: false
+  ad_set_encryption_types:
+    description:
+      - Set this V(true) if creation of Active Directory accounts should automatically turn on the associated encryption types represented by the msDS-EncryptionTypes field.
+      - Used only if O(kdc_type='Active Directory').
+    type: bool
+    required: false
+  kdc_account_creation_host_override:
+    description:
+      - Active Directory Domain Controller host override.
+      - This parameter should be used when multiple Active Directory Domain Controllers are behind a load-balancer.
+      - This parameter should be set with the address of one of them AD Domain Controller.
+      - This setting is used only while creating accounts. CDH services use the value entered in the O(kdc_host) while authenticating.
+      - Only applicable if O(kdc_type='Active Directory')
+    type: str
+    required: false
+  gen_keytab_script:
+    description:
+      - Custom Kerberos Keytab Retrieval Script.
+      - Specify the path to a custom script, or executable, to retrieve a Kerberos keytab.
+      - The script should take two arguments - a destination file to write the keytab to, and the full principal name to retrieve the key for.
```

Comment:
The target script should accept two arguments: a _destination path_ for the resulting keytab and the _full principal name_ of the owner of the keytab.

Reply:
Reworded.